CONTRIBUTING.md

```diff
@@ -5,6 +5,79 @@ Welcome, and thank you for your interest in contributing to iTwin.js!
+
+Each package has its own **node_modules** directory that contains symbolic links to *common* dependencies managed by Rush.
+
+1. Install dependencies: `rush install`
+2. Build source: `rush build`
+3. Run tests: `rush cover`
+
+The above commands iterate and perform their action against each package in the monorepo.
+
+## Source Code Edit Workflow
+
+<details>
+  <summary>Once you've set up the repo, you can move on to making changes to the source code by following the steps within: </summary>
+
+1. Create your own branch by running `git checkout -b "<your-branch-name>"`. See [Branch Naming Policy](#branch-naming-policy)
+2. Make source code changes.
+3. Rebuild the repo by running `rush build`.
+4. Ensure unit tests pass when run locally: `rush cover`.
+5. Ensure linting passes when run locally: `rush lint`.
+6. Locally commit changes: `git commit` (or use the Visual Studio Code user interface).
+7. Repeat steps 1-4 until ready to push changes.
+8. Check for API signature changes: `rush extract-api`. This will update the signature files, located in `common/api`. **Note:** before doing this, first do the following:
+    - Be sure that your branch is up to date with the target branch (i.e. `git merge origin/master`).
+    - Clean up your build output: `rush clean`.
+    - Rebuild the project: `rush build`.
+9. Review any diffs to the API signature files in the `common/api` directory to ensure they are compatible with the intended release of the package.
+    - If any differences are in packages not modified on this branch, revert the changes before committing.
+10. Add changelog entry (which could potentially cover several commits): `rush change`.
+11. Follow prompts to enter a change description or press ENTER if the change does not warrant a changelog entry. If multiple packages have changed, multiple sets of prompts will be presented. If the changes are only to non-published packages (like **display-test-app**), then `rush change` will indicate that a changelog entry is not needed.
+12. Completing the `rush change` prompts will cause new changelog entry JSON files to be created.
+13. Add and commit the changelog JSON files and any API signature updates.
+14. Publish changes on the branch and open a pull request.
+
+> If executing scripts from `package.json` files in any of the subdirectories, we recommend using [`rushx`](https://rushjs.io/pages/commands/rushx/) over `npm`.
+
+If using the command line on Windows, steps 8 through 11 above can be completed in one step by running `rushchange.bat` from the imodeljs root directory.
+Only use `rushchange.bat` if none of the changes require a changelog entry.
+> The CI build will break if changes are pushed without running `rush change` and `rush extract-api` (if the API was changed). The fix will be to complete steps 6 through 11.
+
+Here is a sample [changelog](https://github.com/microsoft/rushstack/blob/master/apps/rush/CHANGELOG.md) to demonstrate the level of detail expected.
```

Comment:
might want to note changelogs are user facing so they should avoid unecessary details

Reply:
added a sentence encouraging less verbosity and unnecessary fluff